### DISCUSSION TITLE
writeBinary Causes unused textures to be lost

### DISCUSSION THREAD

**Discussion by wxxewx** [donmccurdy/glTF-Transform#587] [General]
io.writeBinary(doc)  Causes my unused textures to be lost。These textures are referenced in the extension，Use is VRM

**Comment by donmccurdy** [comment]
`io.writeBinary(doc)` won't remove unused textures — Unless you're also using the [prune()](https://gltf-transform.donmccurdy.com/functions.html#prune) function, unused resources are kept. However, it is possible that the order and indices of textures might change, and if a custom extension references those textures, but isn't [implemented and registered as a custom extension](https://gltf-transform.donmccurdy.com/extensions.html#custom-extensions), then its texture reference index might be wrong.

Without more information I won't be able to investigate further, so please share code models to debug if you believe there is a bug. Thanks!
